controllers/movie.js

```diff
@@ -48,25 +54,34 @@ const readByGenre = async (req, res) => {
 
 const readByRecommendation = async (req, res) => {
   console.log(req.body);
+  const { certifications, genres, release, ratings, providers, foreign } = req.body;
 
-  const applicableCerts = recommendations.getCertifications(
-    req.body.certification,
+  // const applicableCerts = recommendations.getCertifications(req.body.certification);
+  const applicableCerts = Object.keys(certifications).filter(
+    certKey => certifications[certKey],
   );
+  const applicableGenres = Object.keys(genres).filter(genreKey => genres[genreKey]);
 
   const conditions = {
-    genres: { $in: req.body.genres },
+    genres: { $in: applicableGenres },
     certification: { $in: applicableCerts },
-    release_year: { $gte: req.body.min_year, $lte: req.body.max_year },
-    'ratings.rotten_tomatoes.value': { $gte: req.body.rotten_tomatoes },
-    'ratings.imdb.value': { $gte: req.body.imdb },
+    release_year: { $gte: release.minYear, $lte: release.maxYear },
+    'ratings.rotten_tomatoes.value': {
+      $gte: ratings.rottenTomatoes.minRating,
+      $lte: ratings.rottenTomatoes.maxRating,
+    },
+    'ratings.imdb.value': { $gte: ratings.imdb.minRating, $lte: ratings.imdb.maxRating },
   };
-
-  if (!req.body.foreign) conditions.original_language = 'en';
-  if (req.body.indie)
-    conditions.budget = { $lte: recommendations.INDIE_BUDGET_THRESHOLD };
+  if (!foreign) conditions.original_language = 'en';
+  // TODO: add provider checking?
```

Comment:
I'm assuming you are coming back to this? If you are, fine! If not, you can just remove it :)

Reply:
We will potentially come back to this. I talked to Glen about it and based on how the back-end/database is currently structured there isn't really a good way to do this filtering/checking back here so it currently needs to just be dealt on the front end. I'm leaving the comment for now in case we change things such that it's more viable later.